pkg/resources/grant_helpers.go

```diff
@@ -52,24 +53,67 @@ type grant struct {
 	GrantOption bool
 }
 
-// splitGrantID takes the <db_name>|<schema_name>|<view_name>|<privilege> ID and
-// returns the object name and privilege.
-func splitGrantID(v string) (string, string, string, string, error) {
-	reader := csv.NewReader(strings.NewReader(v))
-	reader.Comma = '|'
+// grantID contains identifying elements that allow unique access privileges
+type grantID struct {
+	ResourceName string
+	SchemaName   string
+	ViewName     string
+	Privilege    string
+}
+
+func (grantID *grantID) String() (string, error) {
+	var buf bytes.Buffer
+	csvWriter := csv.NewWriter(&buf)
+	csvWriter.Comma = '|'
+	dataIdentifiers := [][]string{{grantID.ResourceName, grantID.SchemaName, grantID.ViewName, grantID.Privilege}}
+	err := csvWriter.WriteAll(dataIdentifiers)
+
+	if err != nil {
+		return "", err
+	}
+
+	return buf.String(), nil
+}
 
+func grantIDFromString(stringID string) (*grantID, error) {
```

Comment:
this function can probably use a comment like the one `splitGrantID` used to have
```
// splitGrantID takes the <db_name>|<schema_name>|<view_name>|<privilege> ID and
...
```